src/getDocumentCursorStream.js

```diff
@@ -0,0 +1,68 @@
+const {Transform} = require('node:stream')
+
+const pkg = require('../package.json')
+const requestStream = require('./requestStream')
+
+module.exports = async (options) => {
+  let streamsInflight = 0
+  const stream = new Transform({
+    async transform(chunk, encoding, callback) {
+      if (encoding !== 'buffer' && encoding !== 'string') {
+        callback(null, chunk)
+        return
+      }
+
+      try {
+        const parsedChunk = JSON.parse(chunk.toString())
+        if (
+          parsedChunk !== null &&
+          typeof parsedChunk === 'object' &&
+          'nextCursor' in parsedChunk &&
+          typeof parsedChunk.nextCursor === 'string' &&
+          !('_id' in parsedChunk)
+        ) {
+          streamsInflight++
+
+          const reqStream = await startStream(options, parsedChunk.nextCursor)
+          reqStream.on('end', () => {
+            streamsInflight--
+            if (streamsInflight === 0) {
+              stream.end()
+            }
+          })
+          reqStream.pipe(this, {end: false})
+
+          callback()
+          return
+        }
+      } catch (err) {
+        // Ignore JSON parse errors
```

Comment:
Is that wise? Smells like the sort of thing that can cause hard-to-figure-out Node.js problems. Better to be strict in such situations; the stream should _not_ have parse errors.

Reply:
I can move the catch into JSON parse only, it's only if it's sending something that isn't json the error should be handled by a later consumer